addons/l10n_ar_withholding/models/account_tax.py

```diff
@@ -7,13 +7,36 @@ class AccountTax(models.Model):
     _inherit = 'account.tax'
 
     l10n_ar_withholding_payment_type = fields.Selection(
-        [('supplier', 'Supplier'), ('customer', 'Customer')], 'Argentinean Withholding type',
-        compute="_compute_l10n_ar_withholding_payment_type", store=True, readonly=False)
-
+        [('supplier', 'Supplier'), ('customer', 'Customer')], 'Argentinean Withholding Type',
+        compute="_compute_l10n_ar_withholding_payment_type", store=True, readonly=False, help="Withholding tax for supplier or customer payments.")
+    l10n_ar_tax_type = fields.Selection(string='Argentinean Tax Type', selection=[
+        ('earnings_withholding', 'Earnings'),
+        ('earnings_withholding_scale', 'Earnings Scale'),
+        ('iibb_withholding_untaxed', 'IIBB Untaxed'),
+        ('iibb_withholding_total', 'IIBB Total Amount'),
+    ],
+        compute="_compute_l10n_ar_tax_type", store=True, readonly=False)
     l10n_ar_withholding_sequence_id = fields.Many2one(
-        'ir.sequence', 'Withholding Number Sequence', copy=False, check_company=True,
+        'ir.sequence', 'Argentinean Withholding Sequence', copy=False, check_company=True,
         help='If no sequence provided then it will be required for you to enter withholding number when registering one.')
+    l10n_ar_code = fields.Char('Argentinean Code', help="Earning withholding regimen.")
+    l10n_ar_none_taxable_amount = fields.Float(
+        'Argentinean None Taxable Amount',
+        digits='Account',
+        help="If base amount is lower than Argentinean None Taxable Amount, then withholding amount is 0.0. Else withholdable base amount = base amount - Argentinean None Taxable Amount"
+    )
+    l10n_ar_state_id = fields.Many2one(
+        'res.country.state', string="Argentinean State", ondelete='restrict', domain="[('country_id', '=?', country_id)]")
+    l10n_ar_scale_id = fields.Many2one(
+        comodel_name='afip.earnings_table.scale',
+        help="Earnings table scale if tax type is 'Earnings Scale'."
+    )
+    l10n_ar_minimun_withholdable_amount = fields.Float(default=0.0, help="If withholding amount is lower than minimun withholdable amount then withholding amount is 0.0.")
```

Comment:
```suggestion
    l10n_ar_minimum_withholdable_amount = fields.Monetary(default=0.0, help="If withholding amount is lower than minimum withholdable amount then no withholding is applied?")
```
help msg is weird

Reply:
This field is only used to do calculations and I understand that this amount should not be associated with a currency because the taxable amount must be always in ARS (argentinean currency). If a payment has other currency different than ARS we need to convert the taxable amount to amount in argentinean currency to do the calculations. We can talk more about this point in the Follow up meeting. @vbe-odoo

Reply:
add related field (company currency) and change field to monetary

Reply:
keep with the final part of the help message

Reply:
Add "base amount" field on payment view form for each withholding (before amount column) --> withholdings (one2many). Remove withholding with 0.0 amount in the pdf. Examples --> Ecuador and India